File: packages/expo-maps/android/src/main/java/expo/modules/maps/records/LatLngRecord.kt
package expo.modules.maps.records

import com.google.android.gms.maps.model.LatLng
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class LatLngRecord(latLng: LatLng) : Record {
  @Field
  var longitude: Double = latLng.longitude

  @Field
  var latitude: Double = latLng.latitude
}

File: packages/expo-maps/ios/ExpoMaps/Records/LatLngRecord.swift
import CoreLocation
import ExpoModulesCore

struct LatLngRecord: Record {
  init() {}

  @Field var latitude: Double?
  @Field var longitude: Double?

  init(coordinate: CLLocationCoordinate2D) {
    latitude = coordinate.latitude
    longitude = coordinate.longitude
  }
}
